mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -443,12 +425,11 @@ private[ml] trait GBTParams extends TreeEnsembleParams with HasMaxIter with HasS
    * (default = 0.1)
    * @group setParam
    */
-  def setStepSize(value: Double): this.type = set(stepSize, value)
-
-  override def validateParams(): Unit = {
+  def setStepSize(value: Double): this.type = {
     require(ParamValidators.inRange(0, 1, lowerInclusive = false, upperInclusive = true)(
```

Comment:
The original ```validateParams()``` was only used to check for interactions between parameters. Parameter value checks which do not depend on other parameters are handled by ````Param.validate()``` at the definition, i.e.
```
final val k = new IntParam(this, "k", "The number of clusters to create. " +
    "Must be > 1.", ParamValidators.gt(1))
```
However, ```stepSize``` was a trait and inherited by lots of sub-classes which have different constraints for this parameter, so I add the check at setter.